### PR TITLE
Add authorization rules

### PR DESCRIPTION
**Pull request by fithamlak**
`Add authorization rules` requeirements are
-  Install CanCanCan in your project.
-  Add a role column to the users table. Remember to use a migration for this.
-  A user can delete a post if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
   -  For that you need to implement the post deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.
-  A user can delete a comment if it is theirs or if they have an admin role (column role has value "admin"). Use CanCanCan for this authorization.
   -  For that you need to implement the comment deleting functionality. Add the "Delete" button to the view and make sure that only authorized users can see it.

### CODE REVIEW

File: app/models/ability.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+    can :destroy, Comment do |comment|
+      comment.author == user
+    end
+
+    # additional permissions for admin users
+    return unless user.role == 'admin'
+
+    can :destroy, Comment
+    can :destroy, Post
```
Comment (TadesseDev) [inline app/models/ability.rb]: - [Optional] Dear team, this is not the optimal way of allowing user to destroy. we need to explicitly allow permissions by checking the role of the user. something like  
````ruby
 if user.admin? 
         can :manage, :all
````

this makes it explicit what we are trying to accomplish and to read and understand the code. 

- [Optional] Also, I would suggest having this line next to `return unless user.present?`. This allows our program to terminate early, which has a benefit from an efficiency perspective.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -1,10 +1,10 @@
 class PostsController < ApplicationController
+  skip_before_action :authenticate_user!, only: %i[index show]
+  before_action :set_user, only: :index
+  before_action :set_likes, only: :index
+
   def index
```
Comment (TadesseDev) [inline app/controllers/posts_controller.rb]: - [ ] You did a great job team, But we need to use **CanCanCan** for Post authorization in `Post Controller`. 
> something like 
```ruby
  class PostsController < ApplicationController
    load_and_authorize_resource

    ...
  end
```

**Review by TadesseDev — CHANGES_REQUESTED** [review]
### Hi Team

### I'm Tadesse Alemayehu your reviewer for this PR. :hand: 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!
212094426-b41dd457-c9db-4ab3-bbe9-176dd8dd08b7




## Highlights
 - [x]  Correct GitFlow  ✅
 - [x]  Good commit history  ✅
 - [x]  Good PR title/description  ✅
 - [x]  All linters passed  ✅
 
## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/models/ability.rb (hunk 1 above)
Comment (fithamlak) [reply app/models/ability.rb]: Thank you @TadesseDev  senior for your constractive feedback

Comment (fithamlak) [reply app/models/ability.rb]: I try to put all suggetions

**Review by rustam-tolipov — APPROVED** [review]
Hi Team, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Developer-512px-10

## Highlights
1. No Linters errors 💯
2. All requirements are complete 💯
3. Great readme 💯 
4. Well-implemented code 👍 

## Optional suggestions
No optional suggestions :)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**